templates/CRM/Admin/Form/Setting/Search.tpl

```diff
@@ -102,3 +110,15 @@
             <div class="crm-submit-buttons">{include file="CRM/common/formButtons.tpl" location="bottom"}</div>
 
 </div>
+{literal}
+<script type="text/javascript">
+function handleCheckboxUpdate(event) {
+    var checkboxes = $(event.target).find('input'),
+        sortedCheckboxes = [];
+    for (var i = 0; i < checkboxes.length; i++) {
+        sortedCheckboxes.push(checkboxes[i].name.split(/[\[\]]/)[1]);
+    }
+    CRM.$('#sortedQuicksearchOptions').val(sortedCheckboxes);
```

Comment:
so I was wondering, is it really hard for the main quickSearchOptions field to hold the sorting instead of this hidden field ?

Reply:
Yes, I don't think it possible without making changes to the `HTML/QuickForm` package. From `$params = $this->controller->exportValues($this->_name);`, the params come in the order they were declared to Quickform. We could access them in the sorted order directly from `$_POST` but I feel that might be a bit of antipattern.

Reply:
hmmm, well I think getting the correct  order $_POST will make the code easier to reuse instead of for any one who wants to archive sortable check-boxes to create an additional hidden field.

You can get the value of $_POST using this core method : 

**CRM_Utils_Request::retrieve()**

Reply:
I agree. In general we should not be accessing $_POST directly, but creating a low-level function here seems appropriate. It could:

- Only act if the settings metadata say this field is sortable
- Only trust the data coming from the quickform submission, but reorder it according to $_POST